File: socket-communication/src/main/java/com/example/socketcommunication/domain/user/presentation/UserController.java
package com.example.socketcommunication.domain.user.presentation;

import com.example.socketcommunication.domain.user.presentation.dto.request.LoginRequest;
import com.example.socketcommunication.domain.user.presentation.dto.request.SignupRequest;
import com.example.socketcommunication.domain.user.presentation.dto.response.TokenResponse;
import com.example.socketcommunication.domain.user.service.UserSignInService;
import com.example.socketcommunication.domain.user.service.UserSignUpService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

@RequiredArgsConstructor
@RequestMapping("/users")
@RestController
public class UserController {

    private final UserSignUpService userSignUpService;
    private final UserSignInService userSignInService;

    @ResponseStatus(HttpStatus.CREATED)
    @PostMapping
    public TokenResponse Signup(@RequestBody SignupRequest request) {
        return userSignUpService.execute(request);
    }

    @PostMapping("/auth")
    public TokenResponse login(@RequestBody LoginRequest request) {
        return userSignInService.execute(request);
    }

}

File: practice-kotlin/src/main/kotlin/com/example/practicekotlin/domain/user/presentation/UserController.kt
package com.example.practicekotlin.domain.user.presentation

import com.example.practicekotlin.domain.user.presentation.dto.request.UserSignUpRequest
import com.example.practicekotlin.domain.user.service.UserSignUpService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.ResponseStatus
import org.springframework.web.bind.annotation.RestController

@RequestMapping("/users")
@RestController
class UserController(
    private val userSignUpService: UserSignUpService
) {

    @ResponseStatus(HttpStatus.CREATED)
    @PostMapping
    fun userSignUp(@RequestBody request: UserSignUpRequest) {
        userSignUpService.execute(request)
    }

}